walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/datasource/network/apis/HttpAgentApiProvider.kt

```diff
@@ -0,0 +1,22 @@
+package com.microsoft.walletlibrary.did.sdk.datasource.network.apis
+
+import com.microsoft.walletlibrary.did.sdk.util.HttpAgentUtils
+import com.microsoft.walletlibrary.util.http.httpagent.IHttpAgent
+import kotlinx.serialization.json.Json
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+
+@Singleton
+internal class HttpAgentApiProvider @Inject constructor(@Named("agent") agent: IHttpAgent, httpAgentUtils: HttpAgentUtils, json: Json) {
```

Comment:
nit: Add class comments to describe briefly about the functions performed by it.